webapp/src/modules/mortgage/sagas.js

```diff
@@ -0,0 +1,143 @@
+      parcel,
+      amount
+    } = action
+
+    const rcnEngineContract = eth.getContract('RCNEngine')
+    const mortgageCreatorContract = eth.getContract('MortgageCreator')
+    const landRegistryContract = eth.getContract('LANDRegistry')
+    const kyberOrcaleAddress = getKyberOracleAddress()
+    const manaCurrency = eth.utils.fromAscii('MANA')
+
+    const landId = yield call(() =>
+      landRegistryContract.encodeTokenId(parcel.x, parcel.y)
+    )
+    const borrower = yield select(getAddress)
+    const loanMetadata = getLoanMetadata()
+
+    let loanParams = [
+      eth.utils.toWei(amount), // Amount requested
+      toInterestRate(interestRate), // Anual interest
+      toInterestRate(punitoryRate), // Anual punnitory interest
+      daysToSeconds(duration), // Duration of the loan, in seconds
+      daysToSeconds(payableAt), // Time when the payment of the loan starts
+      expiresAt // Expiration timestamp of the request
+    ]
+
+    // Retrieve the loan signature
+    let loanIdentifier = yield call(() =>
+      rcnEngineContract.buildIdentifier(
+        kyberOrcaleAddress, // Contract of the oracle
+        borrower, // Borrower of the loan (caller of this method)
+        mortgageCreatorContract.address, // Creator of the loan, the mortgage creator
+        manaCurrency, // Currency of the loan, MANA
+        ...loanParams,
+        loanMetadata // Metadata
+      )
+    )
+
+    // Sign the loan
+    let { signature } = yield call(() => eth.sign(loanIdentifier))
+    const approveSignature = signature.slice(2)
+    let r = `0x${approveSignature.slice(0, 64)}`
+    let s = `0x${approveSignature.slice(64, 128)}`
+    let v = eth.utils.toDecimal(`0x${approveSignature.slice(128, 130)}`)
+
+    // Request a Mortgage
+    let mortgageReceipt = yield call(() =>
+      mortgageCreatorContract.requestMortgage(
+        loanParams, // Configuration of the loan request
+        loanMetadata, // Metadata of the loan
+        landId, // Id of the loan to buy
+        v, // Signature of the loan
+        r, // Signature of the loan
+        s // Signature of the loan
+      )
+    )
+
+    yield put(createMortgageSuccess(mortgageReceipt, parcel))
+    yield put(push(locations.activity))
+  } catch (error) {
+    yield put(createMortgageFailure(error.message))
+  }
+}
+
+function* handleCancelMortgageRequest(action) {
+  try {
+    const { mortgageId, assetId } = action
+    const mortgageManagerContract = eth.getContract('MortgageManager')
+
+    const mortgageCancelReceipt = yield call(() =>
+      mortgageManagerContract.cancelMortgage(mortgageId)
+    )
+
+    yield put(cancelMortgageSuccess(mortgageCancelReceipt, assetId))
+    yield put(push(locations.activity))
+  } catch (error) {
+    yield put(cancelMortgageFailure(error.message))
+  }
+}
+
+function* handleFetchActiveParcelMortgagesRequest(action) {
+  try {
+    const { x, y } = action
+    const mortgages = yield call(() => api.fetchActiveMortgages(x, y, 'active'))
```

Comment:
Repeating myself but, same comment as the API, I would expect `api.fetchActiveMortgages(x, y)` or `api.fetchMortgages(x, y, 'active')`

Reply:
☝️